bootstrap/secret/insecure_test.go

```diff
@@ -93,3 +93,53 @@ func TestInsecureProvider_GetAccessToken(t *testing.T) {
 	require.NoError(t, err)
 	assert.Len(t, actualToken, 0)
 }
+
+func TestInsecureProvider_HasSecrets(t *testing.T) {
+	configAllSecrets := TestConfig{
+		InsecureSecrets: map[string]bootstrapConfig.InsecureSecretsInfo{
+			"DB": {
+				Path:    expectedPath,
+				Secrets: expectedSecrets,
+			},
+		},
+	}
+
+	configMissingSecrets := TestConfig{
+		InsecureSecrets: map[string]bootstrapConfig.InsecureSecretsInfo{
+			"DB": {
+				Path: "redis",
+			},
+		},
+	}
+
+	configNoSecrets := TestConfig{}
+
+	tests := []struct {
+		Name          string
+		Path          string
+		Keys          []string
+		Config        TestConfig
+		ExpectError   bool
+		ExpectResults bool
+	}{
+		{"Valid", expectedPath, []string{"username", "password"}, configAllSecrets, false, true},
+		{"Valid just path", expectedPath, nil, configAllSecrets, false, true},
+		{"Invalid - No secrets", expectedPath, []string{"username", "password"}, configMissingSecrets, false, false},
```

Comment:
This is a valid response to Has?
```suggestion
		{"Valid - No secrets", expectedPath, []string{"username", "password"}, configMissingSecrets, false, false},
```